server.ts

```diff
@@ -0,0 +1,57 @@
+import * as express from 'express';
+import { TodoApp } from './src/todo';
+
+const todo = new TodoApp();
+
+const serv = express();
+serv.use(express.json());
+
+const port = 3000;
+
+serv.get('/', function (req, res) {
+    res.send(todo.listAllToDos());
+});
+
+serv.put('/', function (req, res) {
+    const data = req.body;
+    try {
+        todo.addItem(data.description, data.dueAt);
+        res.sendStatus(200);
+    } catch (e) {
+        if (e.message = 'The date you entered is invalid') {
+            res.status(400).send('The date you entered is invalid');
+        } else {
+            throw e;
+        }
+    }
+});
+
+serv.post(/^\/[0-9]+\/done/, function (req, res) {
```

Comment:
So there is a way to extract parameters out of an express request more easily.  Look into this.  Search for "express parameters" or something akin to that.